### PR TITLE
Prepare 0.3.0 release

### PR DESCRIPTION
**Pull request by nfx**
azcli
---
 * [x] TestAccClusterResource_CreateClusterWithLibraries (307.67s)
 * [x] TestAccSecretScopeResource (21.15s)
 * [x] access.TestAccPermissionsClusterPolicy (22.97s)
 * [x] access.TestAccPermissionsClusters (315.19s)
 * [x] access.TestAccPermissionsInstancePool (17.70s)
 * [x] access.TestAccPermissionsJobs (18.53s)
 * [x] access.TestAccPermissionsNotebooks (19.33s)
 * [x] access.TestAccPermissionsTokens (15.65s)
 * [x] access/acceptance.TestAccDatabricksPermissionsResourceFullLifecycle (56.90s)
 * [x] access/acceptance.TestAccInitialManagePrincipals (2.76s)
 * [x] access/acceptance.TestAccInitialManagePrincipalsGroup (1.72s)
 * [x] access/acceptance.TestAccSecretAclResource (19.21s)
 * [x] access/acceptance.TestAccSecretAclResourceDefaultPrincipal (14.45s)
 * [x] access/acceptance.TestAccSecretResource (21.20s)
 * [x] access/acceptance.TestAzureAccKeyVaultSimple (3.10s)
 * [x] compute.TestAccContext (54.13s)
 * [x] compute.TestAccInstancePools (6.87s)
 * [x] compute.TestAccLibraryCreate (90.33s)
 * [x] compute.TestAccListClustersIntegration (262.03s)
 * [x] compute.TestAzureAccNodeTypes (3.06s)
 * [x] compute/acceptance.TestAccClusterPolicyResourceFullLifecycle (31.58s)
 * [x] compute/acceptance.TestAccClusterResource_CreateSingleNodeCluster (263.09s)
 * [x] compute/acceptance.TestAccJobResource (21.82s)
 * [x] identity.TestAccCreateToken (3.84s)
 * [x] identity.TestAccCreateUser (9.39s)
 * [x] identity.TestAccFilterGroup (4.13s)
 * [x] identity.TestAccGroup (28.59s)
 * [x] identity.TestAccReadUser (4.45s)
 * [x] identity.TestAzureAccSP (10.97s)
 * [x] identity/acceptance.TestAccGroupMemberResource (29.86s)
 * [x] identity/acceptance.TestAccGroupResource (33.05s)
 * [x] identity/acceptance.TestAccGroupResource_verify_entitlements (23.59s)
 * [x] identity/acceptance.TestAccTokenResource (13.84s)
 * [x] identity/acceptance.TestAccUserResource (25.41s)
 * [x] identity/acceptance.TestAzureAccServicePrincipalResource (28.68s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace (4.79s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Cluster_Policies (0.49s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Cluster_Policies_Delete (0.48s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Clusters (0.31s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/DBFS_Files (0.67s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Groups (0.80s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Instance_Pools (0.22s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Jobs (0.22s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Notebooks (0.20s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Secret_ACLs (0.20s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Secret_Scopes (0.15s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Secrets (0.15s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Tokens (0.53s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Users (0.37s)
 * [x] internal/sanity.TestAccMutiworkspaceUsedFromNormalMode (9.00s)
 * [x] storage.TestAccCreateFile (5.75s)
 * [x] storage.TestAccDeleteInvalidMountFails (7.44s)
 * [x] storage.TestAccInvalidSecretScopeFails (7.35s)
 * [x] storage.TestAccSourceOnInvalidMountFails (30.34s)
 * [x] storage.TestAzureAccBlobMount (150.22s)
 * [x] storage/acceptance.TestAccDatabricksDBFSFile_CreateViaContent (21.70s)
 * [x] storage/acceptance.TestAccDatabricksDBFSFile_CreateViaSource (10.38s)
 * [x] storage/acceptance.TestAzureAccBlobMount_correctly_mounts (675.16s)
 * [x] workspace/acceptance.TestAccNotebookResourceScalability (24.51s)
 * [x] workspace/acceptance.TestAccWorkspaceConfFullLifecycle (11.01s)

awsmt
---
 * [x] TestAccClusterResource_CreateClusterWithLibraries (222.10s)
 * [x] TestAccClusterResource_CreateSingleNodeCluster (196.51s)
 * [x] TestAwsAccGroupInstanceProfileResource (1138.57s)
 * [x] TestAwsAccS3IamMount_WithCluster (1143.30s)
 * [x] access.TestAccIPACL (3.31s)
 * [x] access.TestAccPermissionsClusterPolicy (12.93s)
 * [x] access.TestAccPermissionsClusters (215.48s)
 * [x] access.TestAccPermissionsInstancePool (15.51s)
 * [x] access.TestAccPermissionsJobs (16.82s)
 * [x] access.TestAccPermissionsNotebooks (16.73s)
 * [x] access.TestAccPermissionsTokens (15.20s)
 * [x] access/acceptance.TestAccDatabricksPermissionsResourceFullLifecycle (31.54s)
 * [x] access/acceptance.TestAccInitialManagePrincipals (1.81s)
 * [x] access/acceptance.TestAccInitialManagePrincipalsGroup (0.77s)
 * [x] access/acceptance.TestAccSecretAclResource (13.52s)
 * [x] access/acceptance.TestAccSecretAclResourceDefaultPrincipal (5.29s)
 * [x] access/acceptance.TestAccSecretResource (5.00s)
 * [x] access/acceptance.TestAccSecretScopeResource (6.29s)
 * [x] compute.TestAccContext (110.52s)
 * [x] compute.TestAccInstancePools (2.82s)
 * [x] compute.TestAccLibraryCreate (73.10s)
 * [x] compute.TestAccListClustersIntegration (197.05s)
 * [x] compute.TestAwsAccSmallestNodeType (0.21s)
 * [x] compute/acceptance.TestAccClusterPolicyResourceFullLifecycle (6.93s)
 * [x] compute/acceptance.TestAccJobResource (2.76s)
 * [x] compute/acceptance.TestAwsAccJobResource_NoInstancePool (2.32s)
 * [x] compute/acceptance.TestAwsAccJobsCreate (1.93s)
 * [x] identity.TestAccCreateToken (0.71s)
 * [x] identity.TestAccCreateUser (9.44s)
 * [x] identity.TestAccFilterGroup (1.87s)
 * [x] identity.TestAccGroup (21.56s)
 * [x] identity.TestAccReadUser (1.73s)
 * [x] identity.TestAwsAccInstanceProfiles (1122.07s)
 * [x] identity/acceptance.TestAccGroupResource (13.95s)
 * [x] identity/acceptance.TestAccGroupResource_verify_entitlements (12.52s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace (4.01s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Cluster_Policies (0.44s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Cluster_Policies_Delete (0.43s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Clusters (0.27s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/DBFS_Files (1.20s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Groups (0.13s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Instance_Pools (0.13s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Instance_Profiles (0.12s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Jobs (0.15s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Notebooks (0.15s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Secret_ACLs (0.16s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Secret_Scopes (0.05s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Secrets (0.07s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Tokens (0.55s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Users (0.15s)
 * [x] internal/sanity.TestAccMutiworkspaceUsedFromNormalMode (1.38s)
 * [x] storage.TestAccCreateFile (2.66s)
 * [x] storage.TestAccDeleteInvalidMountFails (4.28s)
 * [x] storage.TestAccInvalidSecretScopeFails (3.23s)
 * [x] storage.TestAccSourceOnInvalidMountFails (32.18s)
 * [x] storage.TestAwsAccS3Mount (335.19s)
 * [x] workspace/acceptance.TestAccNotebookResourceScalability (5.04s)
 * [x] workspace/acceptance.TestAccWorkspaceConfFullLifecycle (2.61s)

mws
---
 * [x] TestMwsAccWorkspaces (113.80s)
 * [x] mws.TestMwsAccCreds (3.52s)
 * [x] mws.TestMwsAccCustomerManagedKeys (3.47s)
 * [x] mws.TestMwsAccLogDelivery (8.28s)
 * [x] mws.TestMwsAccMissingResources (4.19s)
 * [x] mws.TestMwsAccMissingResources/Credential (1.01s)
 * [x] mws.TestMwsAccMissingResources/Customer_Managed_Key (0.79s)
 * [x] mws.TestMwsAccMissingResources/Network (0.74s)
 * [x] mws.TestMwsAccMissingResources/Storage (0.77s)
 * [x] mws.TestMwsAccMissingResources/Workspace (0.88s)
 * [x] mws.TestMwsAccPAS (2.36s)
 * [x] mws.TestMwsAccStorageConfigurations (3.15s)
 * [x] mws.TestMwsAccVPCEndpoint (106.47s)
 * [x] mws.TestMwsAccWorkspace (0.78s)
 * [x] mws/acceptance.TestMwsAccCredentials (6.28s)
 * [x] mws/acceptance.TestMwsAccCustomerManagedKeys (7.43s)
 * [x] mws/acceptance.TestMwsAccLogDelivery (12.69s)
 * [x] mws/acceptance.TestMwsAccNetworks (5.97s)
 * [x] mws/acceptance.TestMwsAccPrivateAccessSettings (5.38s)
 * [x] mws/acceptance.TestMwsAccStorageConfigurations (6.41s)

azsp
---
 * [x] TestAzureAccADLSv1Mount (4.84s)
 * [x] access.TestAccIPACL (12.16s)
 * [x] access.TestAccPermissionsClusterPolicy (15.29s)
 * [x] access.TestAccPermissionsClusters (275.30s)
 * [x] access.TestAccPermissionsInstancePool (16.62s)
 * [x] access.TestAccPermissionsJobs (13.75s)
 * [x] access.TestAccPermissionsNotebooks (15.63s)
 * [x] access.TestAccPermissionsTokens (12.65s)
 * [x] access/acceptance.TestAccDatabricksPermissionsResourceFullLifecycle (48.42s)
 * [x] access/acceptance.TestAccInitialManagePrincipals (1.78s)
 * [x] access/acceptance.TestAccInitialManagePrincipalsGroup (0.86s)
 * [x] access/acceptance.TestAccSecretAclResource (16.82s)
 * [x] access/acceptance.TestAccSecretAclResourceDefaultPrincipal (9.64s)
 * [x] access/acceptance.TestAccSecretResource (15.42s)
 * [x] access/acceptance.TestAccSecretScopeResource (12.23s)
 * [x] compute.TestAccContext (96.49s)
 * [x] compute.TestAccInstancePools (4.15s)
 * [x] compute.TestAccLibraryCreate (66.92s)
 * [x] compute.TestAccListClustersIntegration (261.96s)
 * [x] compute.TestAzureAccNodeTypes (0.72s)
 * [x] compute/acceptance.TestAccClusterPolicyResourceFullLifecycle (26.00s)
 * [x] compute/acceptance.TestAccJobResource (8.92s)
 * [x] identity.TestAccCreateToken (2.97s)
 * [x] identity.TestAccCreateUser (8.79s)
 * [x] identity.TestAccFilterGroup (2.35s)
 * [x] identity.TestAccGroup (28.77s)
 * [x] identity.TestAccReadUser (2.00s)
 * [x] identity.TestAzureAccSP (8.29s)
 * [x] identity/acceptance.TestAccGroupMemberResource (29.76s)
 * [x] identity/acceptance.TestAccGroupResource (25.51s)
 * [x] identity/acceptance.TestAccGroupResource_verify_entitlements (16.03s)
 * [x] identity/acceptance.TestAccTokenResource (10.76s)
 * [x] identity/acceptance.TestAccUserResource (18.16s)
 * [x] identity/acceptance.TestAzureAccServicePrincipalResource (18.79s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace (6.25s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Cluster_Policies (0.96s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Cluster_Policies_Delete (0.36s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Clusters (0.45s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/DBFS_Files (1.21s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Groups (0.84s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Instance_Pools (0.32s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Jobs (0.33s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Notebooks (0.22s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Secret_ACLs (0.31s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Secret_Scopes (0.26s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Secrets (0.26s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Tokens (0.60s)
 * [x] internal/sanity.TestAccMissingResourcesInWorkspace/Users (0.11s)
 * [x] internal/sanity.TestAccMutiworkspaceUsedFromNormalMode (10.86s)
 * [x] storage/acceptance.TestAzureAccAdlsGen1Mount_correctly_mounts (432.83s)
 * [x] workspace/acceptance.TestAccNotebookResourceScalability (18.98s)
 * [x] workspace/acceptance.TestAccWorkspaceConfFullLifecycle (7.89s)

### CODE REVIEW

File: docs/guides/migration-0.3.x.md [hunk 1]
```diff
@@ -1,7 +1,55 @@
 # Migration from 0.2.x to 0.3.x
 
-Certain resources undergone changes in order to ensure consistency with 
+Certain resources undergone changes in order to ensure consistency with REST API and standard expecte Terraform behavior.
```
Comment (alexott) [inline docs/guides/migration-0.3.x.md:3]: expected not expecte

File: docs/guides/migration-0.3.x.md [hunk 2]
```diff
@@ -1,7 +1,55 @@
...
+* Remove `overwrite` attribute. Starting from v0.3.0 it behaves as if it is set to `true`.
+* Remove `mkdirs` attributes. Starting from v0.3.0 it behaves as if it is set to `true`.
+
+## databricks_cluster
+
+* Rewrite `library_jar`, `library_egg`, `library_whl`, `library_pypi`, `library_cran`, and `library_maven` with [library](https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs/resources/cluster#library-configuration-block) configuration block.
+
+## databricks_dbfs_file
+
+* Rename `content` to `content_base64`, as this closer represents actual data within the field and simplifies internal code reusability.
```
Comment (alexott) [inline docs/guides/migration-0.3.x.md:31]: Similarly to notebook write about  overwrite, mkdirs fields. And also about content_b64_md5, validate_remote_file fields

File: CHANGELOG.md [hunk 3]
```diff
@@ -25,7 +23,7 @@
 * Removed deprecated `databricks_scim_group` resource in favor of [databricks_group](https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs/resources/group).
 * Removed deprecated `databricks_default_user_roles` data source in favor of [databricks_group](https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs/data-sources/group#attribute-reference) data source.
 * Removed deprecated `basic_auth` and `azure_auth` provider configuration blocks in favor of [documented authentication methods](https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs).
-* `format`, `overwrite`, and `mkdirs` were removed from `databricks_notebook`. TODO: handle RESOURCE_ALREADY_EXISTS for mkdirs.
+* `format`, `overwrite`, and `mkdirs` were removed from `databricks_notebook`.
```
Comment (alexott) [inline CHANGELOG.md:26]: Highlight that files and notebooks will be always overwritten.
Also overwrite and mkdirs are removed from dbfs file too

File: docs/resources/cluster.md [hunk 4]
```diff
@@ -197,7 +197,9 @@ Attributes are the same as for the `cluster_log_conf` configuration block.
 
 ## aws_attributes
 
-`aws_attributes` optional configuration block contains attributes related to [clusters running on Amazon Web Services](https://docs.databricks.com/clusters/configure.html#aws-configurations). If not specified at cluster creation, a set of default values will be used.
+`aws_attributes` optional configuration block contains attributes related to [clusters running on Amazon Web Services](https://docs.databricks.com/clusters/configure.html#aws-configurations).
+
+-> **Note** *(AWS only)* Please specify empty configuration block (`aws_attributes {}`), even if you're not setting any custom values. This will prevent any resource update issues.
```
Comment (alexott) [inline docs/resources/cluster.md:202]: Can’t we have it as default value?

**Review by alexott — CHANGES_REQUESTED** [review]
It would be useful to highlight some breaking changes

**Comment by alexott** [comment]
Lint is also failed

File: docs/resources/cluster.md (hunk 4 above)
Comment (nfx) [reply docs/resources/cluster.md:202]: we've tried, but for now we treat it as the known issue.

**Review by alexott — APPROVED** [review]
lgtm

**Comment by muralijv** [comment]
Hello Guys 
How do I define a cluster mode or type ( standard or High concurrency)  in Azure Databricks Terraform resource. May I know the property name . Thanks for your help

**Pull request merged**